src/config.rs

```diff
@@ -223,14 +221,41 @@ pub(crate) struct InternalConfig {
 
 impl InternalConfig {
     pub(crate) fn try_from_config(config: Config) -> Result<Self> {
-        let idle_timeout = config.idle_timeout.unwrap_or(DEFAULT_IDLE_TIMEOUT);
+        let default_idle_timeout: IdleTimeout = IdleTimeout::try_from(DEFAULT_IDLE_TIMEOUT)?; // 60s
```

Comment:
DEFAULT_IDLE_TIMEOUT is 18s ?

Reply:
Yeh, that came from @dirvine as a ostensibly a little inside the 20s that a lot of routers might cut off at. I'll add a comment to reflect this.